src/main/java/org/broadinstitute/hellbender/utils/Nucleotide.java

```diff
@@ -14,19 +14,106 @@
  * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
  */
 public enum Nucleotide {
-    A, C, G, T, N, X, INVALID;
 
-    private static final Nucleotide[] baseToValue = new Nucleotide[Byte.MAX_VALUE + 1];
+    // Basic nucleotide codes,
+    // and their one-bit-encoding masks CODE(0xTGCA):
+    A(0b0001),
+    C(0b0010),
+    G(0b0100),
+    T(0b1000),
+
+    // Extended codes:
+    // CODE(included nucs)
+    R(A,G), // Purine
```

Comment:
Nitpicking, but perhaps make the descriptions here a little more consistent in style?  Currently, some are plural, some end in periods, etc.